File: src/utils/auditlog.ts
import { AuditService } from "../services/AuditService";

export interface AuditPayload {
  actorType: "ADMIN" | "TEACHER" | "STUDENT";
  actorId: string;
  actorName: string;
  actorCode: string;

  action:
    | "LOGIN"
    | "LOGOUT"
    | "CREATE"
    | "UPDATE"
    | "DELETE"
    | "DOWNLOAD";

  entityType: string;
  entityId: string;

  description: string;

  changes?: {
    field: string;
    oldValue: any;
    newValue: any;
  }[];
}

export const auditLog = async (payload: AuditPayload) => {
  await AuditService.log({
    ...payload,
    createdAt: new Date().toISOString(),
  });
};

File: src/api/asp-api/SchoolManagementAPI/Models/AuditLog.cs
namespace SchoolManagementAPI.Models;
public class AuditLog
{
    public string Id { get; set; }

    public string ActorType { get; set; }
    public string ActorId { get; set; }
    public string ActorName { get; set; }
    public string ActorCode { get; set; }

    public string Action { get; set; }
    public string EntityType { get; set; }
    public string EntityId { get; set; }

    public string Description { get; set; }

    public DateTime CreatedAt { get; set; }

    public ICollection<AuditChange> Changes { get; set; }
}